### ISSUE TITLE
Installation fails due to Peer Dependencies

### ISSUE THREAD

**Issue by Sljcom** [sanity-io/sanity#3160]
**Describe the bug**
We are using Microsoft Rush in our project along with sanity. We enabled strictPeerDependencies so the installation would fail if any packages does not met the requirements.

It seems as if @sanity/validation has @sanity/client@^2.0.0 as peerDependency and therefore fails as @sanity/client@3.01 is being installed. It is liekly other Sanity pakcages affected by this when testing out without using Microsoft Rush.

**To Reproduce**
Steps to reproduce the behavior with Microsoft Rush:
1. run "npm install -g @microsoft/rush" (5.62.4)
2. run "npm i -g npm@latest" (8.5.1)
3. run "npm i -g @sanity/cli@latest" (2.27.0)
4. run "mkdir Test"
5. run "cd Test"
6. run "rush init"
7. run "mkdir sanity"
8. run "cd sanity"
8. run "sanity init" and name project sanity
9. delete node_modules and yarn.lock
10. Open rush.json
11. set "strictPeerDependencies": true
12. set "projects" [
     {
      "packageName": "sanity",
      "projectFolder": "sanity"
    } 
]
13. run "rush update"
14. Error

Steps to reproduce the behavior without Microsoft Rush:
1. npm i -g npm@latest (8.5.1)
1. npm i -g @sanity/cli@latest
2. sanity init
3. Delete node_modules and yarn.lock
4. npm i --strict-peer-deps

**Expected behavior**

With Microsoft Rush
```
rush update
 ERROR  @sanity/base: @sanity/validation@2.27.0 requires a peer of @sanity/client@^2.0.0 but version 3.1.0 was installed.
```

Without Microsoft Rush
```
npm i --strict-peer-deps
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @sanity/base@2.27.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0" from the root project
npm ERR!   peer react@"^16.9 || ^17" from @sanity/base@2.27.0
npm ERR!   node_modules/@sanity/base
npm ERR!     @sanity/base@"^2.27.0" from the root project
npm ERR!     @sanity/base@"2.27.0" from @sanity/default-layout@2.27.0
npm ERR!     node_modules/@sanity/default-layout
npm ERR!       @sanity/default-layout@"^2.27.0" from the root project
npm ERR!     2 more (@sanity/desk-tool, @sanity/vision)
npm ERR!   17 more (react-dom, styled-components, @sanity/default-layout, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! react-sortable-hoc@"^1.11.0" from @sanity/base@2.27.0
npm ERR! node_modules/@sanity/base
npm ERR!   @sanity/base@"^2.27.0" from the root project
npm ERR!   @sanity/base@"2.27.0" from @sanity/default-layout@2.27.0
npm ERR!   node_modules/@sanity/default-layout
npm ERR!     @sanity/default-layout@"^2.27.0" from the root project
npm ERR!   2 more (@sanity/desk-tool, @sanity/vision)
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0" from react-sortable-hoc@1.11.0
npm ERR!   node_modules/react-sortable-hoc
npm ERR!     react-sortable-hoc@"^1.11.0" from @sanity/base@2.27.0
npm ERR!     node_modules/@sanity/base
npm ERR!       @sanity/base@"^2.27.0" from the root project
npm ERR!       3 more (@sanity/default-layout, @sanity/desk-tool, @sanity/vision)
```

**Screenshots**

**Which versions of Sanity are you using?**

@sanity/cli                   2.27.0 (up to date)
@sanity/base                  2.27.0 (up to date)
@sanity/core                  2.27.0 (up to date)
@sanity/default-layout        2.27.0 (up to date)
@sanity/default-login         2.27.0 (up to date)
@sanity/desk-tool             2.27.0 (up to date)
@sanity/eslint-config-studio   2.0.0 (up to date)
@sanity/vision                2.27.0 (up to date)

**What operating system are you using?**

macOS Monterey Version 12.0.1

**Which versions of Node.js / npm are you running?**

8.5.1

**Comment by geoffreyjball** [comment]
Thank you for reporting this, @Sljcom. This peer dependency was fixed in `v2.28.0`. Please follow up if you're still experiencing issues and we can reopen.

**Comment by LushawnDev** [comment]
I get the same when trying to use npm in my project. Whether its updating or installing or adding a new package. I have to use `--legacy-peer-deps` whenever I use npm. It seems like bad practice... Here's my error:

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: sanity-plugin-dashboard-widget-netlify@2.0.1
npm error Found: styled-components@6.1.12
npm error node_modules/styled-components
npm error   peer styled-components@"^6.1" from @portabletext/editor@1.0.9
npm error   node_modules/@portabletext/editor
npm error     @portabletext/editor@"^1.0.8" from sanity@3.52.2
npm error     node_modules/sanity
npm error       sanity@"^3.52.2" from the root project
npm error       4 more (@sanity/dashboard, ...)
npm error   peer styled-components@"^5.2 || ^6.0.0" from @sanity/dashboard@3.1.6
npm error   node_modules/@sanity/dashboard
npm error     @sanity/dashboard@"^3.1.4" from the root project
npm error     peer @sanity/dashboard@"^3.0.0" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error     node_modules/sanity-plugin-dashboard-widget-netlify
npm error       sanity-plugin-dashboard-widget-netlify@"^2.0.1" from the root project
npm error   8 more (@sanity/ui, @sanity/ui, @sanity/ui, @sanity/vision, ...)
npm error
npm error Could not resolve dependency:
npm error peer styled-components@"^5.2" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error node_modules/sanity-plugin-dashboard-widget-netlify
npm error   sanity-plugin-dashboard-widget-netlify@"^2.0.1" from the root project
npm error
npm error Conflicting peer dependency: styled-components@5.3.11
npm error node_modules/styled-components
npm error   peer styled-components@"^5.2" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error   node_modules/sanity-plugin-dashboard-widget-netlify
npm error     sanity-plugin-dashboard-widget-netlify@"^2.0.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/lucian/.npm/_logs/2024-07-29T14_07_24_213Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/lucian/.npm/_logs/2024-07-29T14_07_24_213Z-debug-0.log
lucian@DESKTOP-8R34816:~/vscode/client-projects/figureeightcreative.com/backend$ npm install styled-components
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-props-stream@1.0.1
npm warn Found: rxjs@7.8.1
npm warn node_modules/rxjs
npm warn   peer rxjs@"^7" from @portabletext/editor@1.0.9
npm warn   node_modules/@portabletext/editor
npm warn     @portabletext/editor@"^1.0.8" from sanity@3.52.2
npm warn     node_modules/sanity
npm warn   13 more (@sanity/bifur-client, @sanity/client, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer rxjs@"^6" from react-props-stream@1.0.1
npm warn node_modules/react-props-stream
npm warn   react-props-stream@"^1.0.1" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm warn   node_modules/sanity-plugin-dashboard-widget-netlify
npm warn
npm warn Conflicting peer dependency: rxjs@6.6.7
npm warn node_modules/rxjs
npm warn   peer rxjs@"^6" from react-props-stream@1.0.1
npm warn   node_modules/react-props-stream
npm warn     react-props-stream@"^1.0.1" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm warn     node_modules/sanity-plugin-dashboard-widget-netlify
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: sanity-plugin-dashboard-widget-netlify@2.0.1
npm error Found: styled-components@6.1.12
npm error node_modules/styled-components
npm error   styled-components@"*" from the root project
npm error   peer styled-components@"^6.1" from @portabletext/editor@1.0.9
npm error   node_modules/@portabletext/editor
npm error     @portabletext/editor@"^1.0.8" from sanity@3.52.2
npm error     node_modules/sanity
npm error       peer sanity@"^3" from @sanity/dashboard@3.1.6
npm error       node_modules/@sanity/dashboard
npm error         peer @sanity/dashboard@"^3.0.0" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error         node_modules/sanity-plugin-dashboard-widget-netlify
npm error         1 more (the root project)
npm error       4 more (sanity-plugin-dashboard-widget-netlify, ...)
npm error   9 more (@sanity/dashboard, @sanity/ui, @sanity/ui, @sanity/ui, ...)
npm error
npm error Could not resolve dependency:
npm error peer styled-components@"^5.2" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error node_modules/sanity-plugin-dashboard-widget-netlify
npm error   sanity-plugin-dashboard-widget-netlify@"^2.0.1" from the root project
npm error
npm error Conflicting peer dependency: styled-components@5.3.11
npm error node_modules/styled-components
npm error   peer styled-components@"^5.2" from sanity-plugin-dashboard-widget-netlify@2.0.1
npm error   node_modules/sanity-plugin-dashboard-widget-netlify
npm error     sanity-plugin-dashboard-widget-netlify@"^2.0.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/lucian/.npm/_logs/2024-07-29T14_07_45_706Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/lucian/.npm/_logs/2024-07-29T14_07_45_706Z-debug-0.log
```
